RoddyCore/src/de/dkfz/roddy/core/ProjectLoader.groovy

```diff
@@ -457,8 +459,26 @@ class ProjectLoader {
                 sb << "  " << projectID << "@" << aID << Constants.ENV_LINESEPARATOR;
             }
             throw new ProjectLoaderException(sb.toString());
-        } else if (analysis.getRuntimeService() == null) {
+        }
+
+        if (analysis.getRuntimeService() == null) {
             throw new ProjectLoaderException("There is no runtime service class set for the selected analysis. This has to be set in either the project configuration or the analysis configuration.");
         }
+
+        // Earliest check for valid input and output directories. If they are not accessible or writeable.
+        // Start with the input directory
+        for (File _dir = analysis.getInputBaseDirectory(); _dir; _dir = _dir.parentFile) {
```

Comment:
Is there no execution context available yet (addErrorEntry)?

Reply:
BTW Does the addErrorEntry not make more sense in the `Roddy` class? We are dealing with globally fatal errors, so why not put it into the top-level? If so this should be refactored before release of 2.4-final.

Reply:
There is no context available at this point.

It might make sense to have an addErrorEntry class but errors on startup are hardly collected and normally lead to a direct application abort.

Reply:
Of course not a class but a method.